### PR TITLE
fix: Return an err `no blob at a given height` in case there is no blob

### PR DESCRIPTION
**Pull request by yarikbratashchuk**
## Summary by CodeRabbit

- **New Features**
  - Introduced improved error handling in the data retrieval method, ensuring users receive an error message when no data is found for the specified height.

- **Tests**
  - Added a new test case to verify the behavior of the data retrieval method when no blobs exist at a specific height.
  - Modified an existing test to change how errors are managed during concurrent read/write operations.

### CODE REVIEW

**Review by tuxcanfly — DISMISSED** [review]
OK.

File: test/dummy.go [hunk 1]
```diff
@@ -82,7 +82,10 @@ func (d *DummyDA) Get(ctx context.Context, ids []da.ID, _ da.Namespace) ([]da.Bl
 func (d *DummyDA) GetIDs(ctx context.Context, height uint64, _ da.Namespace) ([]da.ID, error) {
 	d.mu.Lock()
 	defer d.mu.Unlock()
-	kvps := d.data[height]
+	kvps, ok := d.data[height]
+	if !ok {
+		return nil, errors.New("no blobs at given height")
```
Comment (tzdybal) [inline test/dummy.go:87]: nitpick: turn this error into sentinel value and use it in tests (I hope this doesn't sound like over-engineering :P)

File: test/test_suite.go [hunk 2]
```diff
@@ -133,8 +136,7 @@ func ConcurrentReadWriteTest(t *testing.T, d da.DA) {
 	go func() {
 		defer wg.Done()
 		for i := uint64(1); i <= 100; i++ {
-			_, err := d.GetIDs(ctx, i, []byte{})
-			assert.NoError(t, err)
+			_, _ = d.GetIDs(ctx, i, []byte{})
```
Comment (tzdybal) [inline test/test_suite.go:139]: I would really like to see that only the nitpickly-introduced error is allowed here.

File: test/test_suite.go [hunk 3]
```diff
@@ -148,3 +150,10 @@ func ConcurrentReadWriteTest(t *testing.T, d da.DA) {
 
 	wg.Wait()
 }
+
+// NoBlobsAtHeightTest tests the case when there are no blobs at a given height in DA
+func NoBlobsAtHeightTest(t *testing.T, d da.DA) {
+	ctx := context.TODO()
+	_, err := d.GetIDs(ctx, 888888, []byte{})
+	assert.Error(t, err)
```
Comment (tzdybal) [inline test/test_suite.go:158]: And here the test could be a bit more specific.

**Review by Manav-Aggarwal — DISMISSED** [review]

File: test/dummy.go (hunk 1 above)
Comment (yarikbratashchuk) [reply test/dummy.go:87]: was referencing to other inline errors tbh
will make it sentinel yes

File: test/test_suite.go (hunk 2 above)
Comment (yarikbratashchuk) [reply test/test_suite.go:139]: fair

File: test/test_suite.go (hunk 3 above)
Comment (yarikbratashchuk) [reply test/test_suite.go:158]: will update

**Comment by yarikbratashchuk** [comment]
@tzdybal as these same fixtures used for proxy test and for direct test, i found that the easiest is to work with strings to identify specific error

**Review by tzdybal — DISMISSED** [review]
:+1:

**Review by tuxcanfly — DISMISSED** [review]
OK.

**Review by tuxcanfly — APPROVED** [review]

**Review by tzdybal — APPROVED** [review]

**Pull request merged**